### PR TITLE
feat: add Resolution API

### LINKED ISSUE #151: IATP: Create `PresentationApi|Controller`
Implement basic glue code to expose the Resolution API. That includes:

- An interface to carry OpenAPI annotations
- A controller class (delegates to the [verifier](#43-subtask-identityhub-implement-titverifier-working-title) and the [query resolver](#42-subtask-identityhub-implement-presentationqueryresolver))
- Model class `PresentationQuery` + `JsonObjectToPresentationQueryTransformer`
- Model classes for VP and VC are implemented here https://github.com/eclipse-edc/Connector/issues/3501

### PR DESCRIPTION
**Pull request by paullatzelsperger**
## What this PR changes/adds

This PR adds the Resolution API for a credential service, that exposes a query endpoint to fetch VerifiableCredentials.

Specifically, its adds the following things:
- A `PresentationApi` + `-Controller`
- A `PresentationQuery` object plus a `JsonObjectToPresentationQueryTransformer`
- Model classes for `presentation_definition` and `PresentationResponse`
- Stubs for collaborator objects: `CredentialQueryResolver`, `AccessTokenVerifier`, `PresentationGenerator` (implementations for those will follow in subsequent PRs).
- an End-To-End test
## Why it does that

Implementation of the `CredentialService`

## Further notes

- This PR removes **most of the existing code base**, as it seemed unpractical to maintain two implementations in the same branch. The DWN-based code can be found in a branch [identity_dwn](https://github.com/eclipse-edc/IdentityHub/tree/identity_dwn)
- some of the classes (`PresentationQuery`, DIF PresEx model classes...) should go into the connector repo eventually
- The huge changeset stems from all the removals

## Linked Issue(s)

Closes #151

### CODE REVIEW

**Review by wolf4ood — APPROVED** [review]
:axe:

File: core/identity-hub-api/src/main/java/org/eclipse/edc/identityservice/api/PresentationApiExtension.java [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+        webService.registerResource(RESOLUTION_CONTEXT, controller);
+
+        // register transformer
+        typeTransformer.register(new JsonObjectToPresentationQueryTransformer(jsonLdMapper));
+        typeTransformer.register(new JsonValueToGenericTypeTransformer(jsonLdMapper));
+    }
+
+    private void cacheContextDocuments() {
+        try {
+            jsonLd.registerCachedDocument("https://identity.foundation/presentation-exchange/submission/v1", Thread.currentThread().getContextClassLoader().getResource("presentation-exchange.v1.json").toURI());
```
Comment (jimmarino) [inline core/identity-hub-api/src/main/java/org/eclipse/edc/identityservice/api/PresentationApiExtension.java:90]: We should use the classloader of the class, not the TCCL as that may be defined differently in different contexts

Also, can we make these constants and introduce a var for the classloader?

File: core/identity-hub-api/src/main/java/org/eclipse/edc/identityservice/api/PresentationApiExtension.java [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+import org.eclipse.edc.validator.spi.JsonObjectValidatorRegistry;
+import org.eclipse.edc.web.jersey.jsonld.JerseyJsonLdInterceptor;
+import org.eclipse.edc.web.jersey.jsonld.ObjectMapperProvider;
+import org.eclipse.edc.web.spi.WebService;
+
+import java.net.URISyntaxException;
+
+import static org.eclipse.edc.spi.CoreConstants.JSON_LD;
+
+public class PresentationApiExtension implements ServiceExtension {
```
Comment (jimmarino) [inline core/identity-hub-api/src/main/java/org/eclipse/edc/identityservice/api/PresentationApiExtension.java:40]: @Extension annotation?

File: core/identity-hub-transform/src/main/java/org/eclipse/edc/identityhub/transform/JsonObjectToPresentationQueryTransformer.java [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+import jakarta.json.JsonValue;
+import org.eclipse.edc.identityhub.spi.model.PresentationQuery;
+import org.eclipse.edc.identityhub.spi.model.presentationdefinition.PresentationDefinition;
+import org.eclipse.edc.jsonld.spi.JsonLdKeywords;
+import org.eclipse.edc.jsonld.spi.transformer.AbstractJsonLdTransformer;
+import org.eclipse.edc.transform.spi.TransformerContext;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+public class JsonObjectToPresentationQueryTransformer extends AbstractJsonLdTransformer<JsonObject, PresentationQuery> {
```
Comment (jimmarino) [inline core/identity-hub-transform/src/main/java/org/eclipse/edc/identityhub/transform/JsonObjectToPresentationQueryTransformer.java:30]: doc

File: e2e-tests/api-tests/src/test/java/org/eclipse/edc/identityhub/tests/fixtures/IdentityHubParticipant.java [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+
+import java.net.URI;
+import java.util.HashMap;
+import java.util.Map;
+
+import static io.restassured.RestAssured.given;
+import static org.eclipse.edc.junit.testfixtures.TestUtils.getFreePort;
+import static org.eclipse.edc.spi.system.ServiceExtensionContext.PARTICIPANT_ID;
+
+public class IdentityHubParticipant {
```
Comment (jimmarino) [inline e2e-tests/api-tests/src/test/java/org/eclipse/edc/identityhub/tests/fixtures/IdentityHubParticipant.java:27]: doc

**Review by jimmarino — CHANGES_REQUESTED** [review]
A few minor comments

File: core/identity-hub-api/src/main/java/org/eclipse/edc/identityservice/api/PresentationApiExtension.java (hunk 1 above)
Comment (paullatzelsperger) [reply core/identity-hub-api/src/main/java/org/eclipse/edc/identityservice/api/PresentationApiExtension.java:90]: yeah, this was copy-pasted from test code :/

**Pull request merged**